fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBExceptionsTest.java

```diff
@@ -65,6 +77,68 @@ private void testWrappedStackTrace(Exception cause, String methodName) {
         ));
     }
 
+    @Test
+    public void tooManySuppressedExceptions() {
+        CompletionExceptionLogHelper.forceSetMaxSuppressedCountForTesting(2);
+        final Exception base = createRuntimeException();
+        Exception e0 = createCompletionException(base);
+        Exception e1 = createCompletionException(base);
+        assertEquals(base, FDBExceptions.wrapException(e0));
+        assertEquals(base, FDBExceptions.wrapException(e1));
+
+        Throwable[] suppressedExceptions = base.getSuppressed();
+        assertEquals(2, suppressedExceptions.length);
+        assertThat(Arrays.asList(suppressedExceptions), contains(e0, e1));
+
+        Exception e2 = createCompletionException(base);
+        assertEquals(base, FDBExceptions.wrapException(e2));
+        suppressedExceptions = base.getSuppressed();
+        assertEquals(3, suppressedExceptions.length);
+        assertThat(Arrays.asList(suppressedExceptions), hasItems(e0, e1));
+        assertThat(Arrays.asList(suppressedExceptions), not(hasItem(e2)));
+        Throwable countException = suppressedExceptions[2];
+        assertThat(countException, instanceOf(CompletionExceptionLogHelper.IgnoredSuppressedExceptionCount.class));
+        assertEquals(1, ((CompletionExceptionLogHelper.IgnoredSuppressedExceptionCount) countException).getCount());
+
+        Exception e3 = createCompletionException(base);
+        assertEquals(base, FDBExceptions.wrapException(e3));
+        suppressedExceptions = base.getSuppressed();
+        assertEquals(3, suppressedExceptions.length);
+        assertThat(Arrays.asList(suppressedExceptions), hasItems(e0, e1));
+        assertThat(Arrays.asList(suppressedExceptions), not(anyOf(hasItem(e2), hasItem(e3))));
+        countException = suppressedExceptions[2];
+        assertThat(countException, instanceOf(CompletionExceptionLogHelper.IgnoredSuppressedExceptionCount.class));
+        assertEquals(2, ((CompletionExceptionLogHelper.IgnoredSuppressedExceptionCount) countException).getCount());
+
+        CompletionExceptionLogHelper.forceSetMaxSuppressedCountForTesting(Integer.MAX_VALUE); //cleanup
```

Comment:
I suppose, in theory, this should be in a `finally` block.

Reply:
Sure, why not.